libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -87,32 +69,63 @@ class PredictiveCacheController @JvmOverloads constructor(
     }
 
     /**
-     * Call when a new map instance is available. Only one map instance at a time is supported.
+     * Create cache controllers for a map instance.
+     * Call when a new map instance is available.
+     *
+     * @param map an instance of [MapboxMap]
+     * @param cacheLocationOptions options to configure cache controllers
+     * @param sourceIdsToCache a list of sources to cache. If empty all available style sources will
+     * be cached
      */
-    fun setMapInstance(map: MapboxMap) {
-        removeMapInstance()
+    @JvmOverloads
+    fun createMapControllers(
+        map: MapboxMap,
+        cacheLocationOptions: PredictiveCacheLocationOptions =
+            PredictiveCacheLocationOptions.Builder().build(),
+        sourceIdsToCache: List<String> = emptyList()
```

Comment:
```suggestion
        sourceIdsToCache: List<String> = getAllMapSources(map)
```
together with
```kotlin
private fun getAllMapSources(map: MapboxMap): List<String> {
    return map.getStyle()?.styleSources
        ?.filter { it.type == VECTOR_SOURCE_TYPE || it.type == RASTER_SOURCE_TYPE }
        ?.map { it.id }
        ?: emptyList()
}
```

This way we don't have to say that "empty means all" but we actually set the right value right away.

Reply:
Hmm, this might not be a good idea. It won't work for style changes.

Reply:
yep, tests don't work
issue, is that we get all sources for the `current` style and they are cached by the `onStyleLoadedListener`
so, if the callback is fired, it will not recalculate ids from a new map, it will use old ones